modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/compute/ClientComputeExecuteMapReduceRequest.java

```diff
@@ -82,11 +85,17 @@ static void packJobIds(ClientMessagePacker out, List<UUID> ids) {
     }
 
     static CompletableFuture<Object> sendTaskResult(TaskExecution<Object> execution, NotificationSender notificationSender) {
+        TaskExecution<Object> t = execution;
         return execution.resultAsync().whenComplete((val, err) ->
-                execution.stateAsync().whenComplete((state, errState) ->
+                t.stateAsync().whenComplete((state, errState) ->
                         execution.statesAsync().whenComplete((states, errStates) ->
                                 notificationSender.sendNotification(w -> {
-                                    w.packObjectAsBinaryTuple(val);
+                                    if (t instanceof MarshallerProvider) {
+                                        Marshaller<Object, byte[]> resultMarshaller = ((MarshallerProvider<Object>) t).resultMarshaller();
+                                        ClientComputeJobPacker.packJobResult(val, resultMarshaller, w);
+                                    } else {
+                                        throw new IllegalArgumentException(t.getClass() + " does not implement MarshallerProvider");
+                                    }
```

Comment:
```suggestion
                                        Marshaller<Object, byte[]> resultMarshaller = ((MarshallerProvider<Object>) t).resultMarshaller();
                                        ClientComputeJobPacker.packJobResult(val, resultMarshaller, w);
```

No need to replace `ClassCastException` with our own logic.